torchgeo/datasets/idtrees.py

```diff
@@ -0,0 +1,553 @@
+        """
+        import laspy
+
+        las = laspy.read(path)
+        array = np.stack([las.x, las.y, las.z], axis=0)
+        tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+        return tensor
+
+    def _load_boxes(self, path: str) -> Tensor:
+        """Load object bounding boxes.
+
+        Args:
+            path: path to .tif file
+
+        Returns:
+            the bounding boxes
+        """
+        base_path = os.path.basename(path)
+
+        # Find object ids and geometries
+        if self.split == "train":
+            indices = self.labels["rsFile"] == base_path
+            ids = self.labels[indices]["id"].tolist()
+            geoms = [self.geometries[i]["geometry"]["coordinates"][0][:4] for i in ids]
+        # Test set - Task 2 has no mapping csv. Mapping is inside of geometry
+        else:
+            ids = [
+                k
+                for k, v in self.geometries.items()
+                if v["properties"]["plotID"] == base_path
+            ]
+            geoms = [self.geometries[i]["geometry"]["coordinates"][0][:4] for i in ids]
+
+        # Convert to pixel coords
+        boxes = []
+        with rasterio.open(path) as f:
+            for geom in geoms:
+                coords = [f.index(x, y) for x, y in geom]
+                xmin = min([coord[0] for coord in coords])
+                xmax = max([coord[0] for coord in coords])
+                ymin = min([coord[1] for coord in coords])
+                ymax = max([coord[1] for coord in coords])
+                boxes.append([xmin, ymin, xmax, ymax])
+
+        tensor: Tensor = torch.tensor(boxes)  # type: ignore[attr-defined]
+        return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load target label for a single sample.
+
+        Args:
+            path: path to image
+
+        Returns:
+            the label
+        """
+        # Find indices for objects in the image
+        base_path = os.path.basename(path)
+        indices = self.labels["rsFile"] == base_path
+
+        # Load object labels
+        classes = self.labels[indices]["taxonID"].tolist()
+        labels = [self.class2idx[c] for c in classes]
+        tensor: Tensor = torch.tensor(labels)  # type: ignore[attr-defined]
+        return tensor
+
+    def _load(self, root: str) -> Tuple[List[str], Dict[int, Dict[str, Any]], Any]:
+        """Load files, geometries, and labels.
+
+        Args:
+            root: root directory
+
+        Returns:
+            the image path, geometries, and labels
+        """
+        import pandas as pd
+
+        if self.split == "train":
+            directory = os.path.join(root, self.directories[self.split][0])
+            labels: pd.DataFrame = self._load_labels(directory)
+            geoms = self._load_geometries(directory)
+        else:
+            directory = os.path.join(root, self.task)
+            if self.task == "task1":
+                geoms = None  # type: ignore[assignment]
```

Comment:
While debugging the recent mypy issues I noticed a lot of typing errors in this file. For example, the `_load` function says that it returns `Tuple[List[str], Dict[int, Dict[str, Any]], Any]`, but many of these return values can also be `None`. The `type: ignore` here is masking this for `geoms`, and the `Any` is masking this for `labels`, but the real issue is that the function is not typed correctly. @isaaccorley can you fix this? We should be ignoring typing errors very sparingly for issues that are beyond our control (PyTorch is missing type hints for large parts of the library), not for things like this. For the same reason, we should try to avoid `Any` as it basically turns off all type checking.

Reply:
I'll refactor but so you know, this dataset has no structure to the nested dictionaries whatsoever so I opted for aDict[str, Any]a in most cases. Also some of the methods return `Any` due to them returning an open3d object or a pandas DataFrame but I'm not able to use those types because I'm not importing them at the top level, only inside the method.

Reply:
See https://mypy.readthedocs.io/en/stable/runtime_troubles.html, it may be possible to declare the type in a string or comment instead to avoid the top-level import.

Reply:
Nice, ok I'll submit a PR.